lib/core/api.js

```diff
@@ -206,20 +208,31 @@ function postBlockToPlatform(data, groupId, callback) {
       callback(jqxhr.responseJSON, 0);
     }
   };
-
-  tidepool.uploadDeviceDataForUser(data, groupId, function(err, result) {
-    if (err) {
-      return sad(err, err);
-    }
-    return happy(result);
-  });
+  if(!_.isEmpty(groupId)){
+    api.log('postBlockToPlatform: for other user');
+    tidepool.uploadDeviceDataForUser(data, groupId, function(err, result) {
```

Comment:
I think the ship has sailed at least for now; I don't have a very strong opinion here and would like to see if @jh-bate does, but I hope this branch will be merged before he gets back from his weekend. If you like, maybe create a Trello card so we don't forget to at least discuss it?